docs/rfcs/1005-realm-key-rotation.md

```diff
@@ -0,0 +1,821 @@
+            },
+            {
+                // `certificate_index` argument doesn't correspond to a `RealmRotateKeyCertificate`
+                "status": "not_found"
+            }
+        ]
+    }
+]
+```
+
+As stated before, client is expected to call `realm_get_keys_bundle` the first time it
+needs to fetch/upload vlob to/from the server, and to store the resulting keys in RAM.
+
+This brings three benefits:
+
+- Keys are no longer stored in each user manifest, which save storage space and avoid
+  having to sync a new version of the user manifest each time a key rotation is done.
+- When access is removed, the user is no longer able to get back the decryption keys
+  (instead of having them in it user manifest). The client can of course save them when
+  it has access to them, but it requires a modified Parsec client.
+- The decryption keys are no longer at rest on the client storage.
+
+## 4 Changes: Data
+
+### 4.1 Existing data: messages
+
+All the messages types are removed:
+
+- `sharing.granted`
+- `sharing.reencrypted`
+- `sharing.revoked`
+
+On top of that, a special logic is added in the client to ignore those messages
+types.
+
+> **Note:**
+>
+> Ignoring those messages means the client may end up with invalid data (e.g. a
+> user hasn't started it client in a while, he does it after the Parsec v3 migration
+> and now all the workspaces it has been part of that have been re-encrypted in the
+> meantime fail to load with an "decryption error" error message...).
+>
+> However those error can simply be fixed by removing and re-adding the access rights
+> to the user, which is the obvious "IT, have you tried to turn it off and on again ?" way.
+
+### 4.2 Existing data: `UserManifest`
+
+Remove `last_processed_message` field.
+Modify `workspaces` field, i.e. the `WorkspaceEntry` nested type:
+
+```json5
+{
+    "label": "WorkspaceEntry",
+    "fields": [
+        {
+            "name": "id",
+            "type": "VlobID"
+        },
+        {
+            "name": "name",
+            "type": "EntryName"
+        }
+        // Removed fields:
+        // - key
+        // - encryption_revision
+        // - encrypted_on
+        // - role_cached_on
+        // - role_cache_value
+    ]
+}
+```
+
+> **Note:**
+>
+> `role_cached_on` and `role_cache_value` are not directly related to this RFC,
+> but given they are no longer in use in Parsec v3, we also remove them.
+
+> **TODO:**
+>
+> Should we rename workspace name into label ?
+>
+> We have "Human handle label" and "Device label", the "label" refers to the
+> fact they are purely display information (i.e. not used for unique identification).
+>
+> So given the idea is similar here (this field can even be modified by the user !),
+> it makes sens to rename it to "workspace label".
+>
+> On the other hand keeping "workspace name" is less work, and the fact this name
+> can be modified where the human handle / device label cannot may justify this
+> is a different concept...
```

Comment:
TODO

Reply:
"workspace name" works differently than human handle label & device label:
- human handle / device label cannot be updated
- human handle / device label are shared among all organization users

hence the renaming would bring confusion, so it's a no !